apps/ProductManagement/Backend/Services/StockService.cs

```diff
@@ -176,4 +176,51 @@ public StockDTO UpdateStock(StockWithoutIDDTO stock, Guid stockId)
 
         return _mapper.Map<StockDTO>((updatedStock, _productDao.Read(stock.ProductId), categories));
     }
+    
+    public List<StockDTO> UpdateStocks(List<StockDTO> stocks)
+    {
+        var updatedStocks = new List<StockDTO>();
+        foreach (var stock in stocks)
+        {
+            var updatedStockEntity = _mapper.Map<Stock>((stock, stock.StockId));
+            _stockDao.Update(updatedStockEntity);
+
+            var updatedStock = _stockDao.Read(stock.StockId);
+            if (updatedStock != null)
+            {
+                List<Category> categories = new List<Category>();
+                var matches = _productCategoriesDao.GetProductCategoriesByProductId(stock.ProductId);
+                foreach (ProductCategories match in matches)
+                {
+                    var category = _categoryDao.Read(match.CategoryId);
+                    if (category != null)
+                    {
+                        categories.Add(category);
+                    }
+                }
+                var stockDTO = _mapper.Map<StockDTO>((updatedStock, _productDao.Read(stock.ProductId), categories));
+                updatedStocks.Add(stockDTO);
+            }
+        }
+        return updatedStocks;
+    }
+    
+    public StockDTO? GetStocksBySubsidiaryAndProductCode(Guid subsidiaryId, int productCode)
+    {
+        var stock = _stockDao.ReadAll().FirstOrDefault(s => s.SubsidiaryId == subsidiaryId && s.Code == productCode);
+        var product = _productDao.Read(stock.ProductId);
+        List<Category> categories = new List<Category>();
+        if (stock != null)
```

Comment:
this validation should be executed before using it to search the product with its content

Reply:
interesting comment I see that for my new fixes